crates/flux-fhir-analysis/src/wf/sortck.rs

```diff
@@ -394,9 +397,34 @@ impl<'genv, 'tcx> InferCtxt<'genv, 'tcx> {
 
                 Ok(sort)
             }
+            fhir::BinOp::BitAnd
+            | fhir::BinOp::BitOr
+            | fhir::BinOp::BitShl
+            | fhir::BinOp::BitShr => self.synth_bitvec_op(expr, op, e1, e2),
         }
     }
 
+    fn synth_bitvec_op(
+        &mut self,
+        expr: &fhir::Expr,
+        op: fhir::BinOp,
+        e1: &fhir::Expr,
+        e2: &fhir::Expr,
+    ) -> Result<rty::Sort> {
+        let name = match op {
+            fhir::BinOp::BitAnd => "bv_and",
+            fhir::BinOp::BitOr => "bv_or",
+            fhir::BinOp::BitShl => "bv_shl",
+            fhir::BinOp::BitShr => "bv_lshr",
+            _ => span_bug!(expr.span, "unexpected operator in synth_bitvec_op"),
+        };
+        let name = crate::Symbol::intern(name);
+        let poly_fsort = flux_middle::THEORY_FUNCS.get(&name).unwrap().sort.clone();
+        let fsort = self.instantiate_func_sort(poly_fsort);
+        let args = vec![e1.clone(), e2.clone()];
```

Comment:
Eventually, I want to move away from having bit vector operations as functions parametric on sizes and have dedicated rules for them. Smtlib is not really parametric on sizes, and we can't express the behavior of more complicated operations like `sign_extend` as function types. I'd prefer if we "inline" the function type as a rule here. But also I don't feel strongly about this.

Reply:
interesting, but I think that's probably best left for a future PR. (We'd probably want to _deprecate_ the `bv_and` etc. functions too then...)